packages/react-devtools-shared/src/backend/console.js

```diff
@@ -136,166 +160,205 @@ export function patch({
-          if (current != null) {
-            try {
-              if (typeof getIsStrictMode === 'function' && getIsStrictMode()) {
-                isInStrictMode = true;
-              }
 
-              if (shouldShowInlineWarningsAndErrors) {
-                // patch() is called by two places: (1) the hook and (2) the renderer backend.
-                // The backend is what implements a message queue, so it's the only one that injects onErrorOrWarning.
-                if (typeof onErrorOrWarning === 'function') {
-                  onErrorOrWarning(
-                    current,
-                    ((method: any): 'error' | 'warn'),
-                    // Copy args before we mutate them (e.g. adding the component stack)
-                    args.slice(),
-                  );
+          const shouldShowInlineWarningsAndErrors =
+            consoleSettingsRef.showInlineWarningsAndErrors &&
+            (method === 'error' || method === 'warn');
+
+          // Search for the first renderer that has a current Fiber.
+          // We don't handle the edge case of stacks for more than one (e.g. interleaved renderers?)
+          // eslint-disable-next-line no-for-of-loops/no-for-of-loops
+          for (const {
+            currentDispatcherRef,
+            getCurrentFiber,
+            onErrorOrWarning,
+            workTagMap,
+          } of injectedRenderers.values()) {
+            const current: ?Fiber = getCurrentFiber();
+            if (current != null) {
+              try {
+                if (shouldShowInlineWarningsAndErrors) {
+                  // patch() is called by two places: (1) the hook and (2) the renderer backend.
+                  // The backend is what implements a message queue, so it's the only one that injects onErrorOrWarning.
+                  if (typeof onErrorOrWarning === 'function') {
+                    onErrorOrWarning(
+                      current,
+                      ((method: any): 'error' | 'warn'),
+                      // Copy args before we mutate them (e.g. adding the component stack)
+                      args.slice(),
+                    );
+                  }
                 }
-              }
 
-              if (shouldAppendWarningStack) {
-                const componentStack = getStackByFiberInDevAndProd(
-                  workTagMap,
-                  current,
-                  currentDispatcherRef,
-                );
-                if (componentStack !== '') {
-                  args.push(componentStack);
+                if (shouldAppendWarningStack) {
+                  const componentStack = getStackByFiberInDevAndProd(
+                    workTagMap,
+                    current,
+                    currentDispatcherRef,
+                  );
+                  if (componentStack !== '') {
+                    if (isStrictModeOverride(args, method)) {
+                      args[0] = format(args[0], componentStack);
```

Comment:
Do we have a test that covers this combo? (Strict mode dimming and console log string substitution)